sdk/core/Azure.Core/tests/HttpPipelineMessageTest.cs

```diff
@@ -100,5 +102,33 @@ public void ContentPropertyThrowsResponseIsExtracted()
             Assert.AreSame(memoryStream, stream);
             Assert.Throws<InvalidOperationException>(() => { var x = response.Content; });
         }
+
+        [Test]
+        public async Task UnbufferedStreamAccessibleAfterMessageDisposed()
+        {
+            var streamBytes = new byte[0x1000];
+            new Random().NextBytes(streamBytes);
+
+            HttpPipeline pipeline = HttpPipelineBuilder.Build(new TestClientOptions() { Retry = { NetworkTimeout = Timeout.InfiniteTimeSpan } });
+
+            using TestServer testServer = new(async context =>
+            {
+                await context.Response.Body.WriteAsync(streamBytes, 0, streamBytes.Length).ConfigureAwait(false);
+            });
+
+            Response response;
+            using (HttpMessage message = pipeline.CreateMessage())
+            {
+                message.Request.Uri.Reset(testServer.Address);
+                message.BufferResponse = false;
+
+                await pipeline.SendAsync(message, default).ConfigureAwait(false);
+                response = message.Response;
+                response.ContentStream = message.ExtractResponseContent();
+            }
+
+            Assert.NotNull(response.ContentStream);
+            Assert.AreEqual(streamBytes[0], response.ContentStream.ReadByte());
```

Comment:
We were previously checking .Length here, but the returned stream is not seekable.